src/ManageCourses.Api/Services/EnrichmentService.cs

```diff
@@ -122,15 +122,18 @@ public bool PublishProviderEnrichment(string providerCode, string email)
                 .OrderByDescending(x => x.Id)
                 .FirstOrDefault();
 
+            var provider = _context.Providers.Single(p => p.ProviderCode == providerCode);
+            var publishTimestamp = DateTime.UtcNow;
+            provider.LastPublishedAt = publishTimestamp;
             if (enrichmentDraftRecord != null)
             {
                 enrichmentDraftRecord.UpdatedAt = DateTime.UtcNow;
                 enrichmentDraftRecord.UpdatedByUser = userOrg.User;
-                enrichmentDraftRecord.LastPublishedAt = DateTime.UtcNow;
+                enrichmentDraftRecord.LastPublishedAt = publishTimestamp;
                 enrichmentDraftRecord.Status = EnumStatus.Published;
-                _context.Save();
                 returnBool = true;
             }
+            _context.Save();
```

Comment:
So regardless if there is a draft enrichment when nothing is published for provider enrichment update provider last published at date?

Reply:
if that is so we can backfill the last published at date, but only if there is a published enrichment